Generator/Sources/NeedleFramework/Generating/Pluginized/PluginizedDependencyProviderContentTask.swift

```diff
@@ -0,0 +1,143 @@
+//
+
+import Foundation
+
+class PluginizedDependencyProviderContentTask: AbstractTask<[PluginizedProcessedDependencyProvider]> {
+
+    /// Initializer.
+    ///
+    /// - parameter providers: The list of providers that we need to fill in
+    /// - parameter providers: The list of pluginized components
+    init(providers: [DependencyProvider], pluginizedComponents: [PluginizedComponent]) {
+        self.providers = providers
+
+        var nonCoreComponentMap = [String: AuxillaryProperties]()
+        var pluginExtensionMap = [String: AuxillaryProperties]()
+        for pluginizedComponent in pluginizedComponents {
+            nonCoreComponentMap[pluginizedComponent.data.name] = AuxillaryProperties(sourceName: pluginizedComponent.nonCoreComponent.name, properties: pluginizedComponent.nonCoreComponent.properties)
+            pluginExtensionMap[pluginizedComponent.data.name] = AuxillaryProperties(sourceName: pluginizedComponent.pluginExtension.name, properties:pluginizedComponent.pluginExtension.properties)
+        }
+        self.nonCoreComponentMap = nonCoreComponentMap
+        self.pluginExtensionMap = pluginExtensionMap
+
+        nonCoreComponentNames = Set(pluginizedComponents.map { pluginizedComponent in
+            pluginizedComponent.nonCoreComponent.name
+        })
+    }
+
+    /// Execute the task and returns the processed in-memory dependency graph
+    /// data models.
+    ///
+    /// - returns: The list of `ProcessedDependencyProvider`.
+    override func execute() -> [PluginizedProcessedDependencyProvider] {
+
+        return providers.map { (provider: DependencyProvider) -> PluginizedProcessedDependencyProvider in
+            do {
+                if provider.pathContains(anyOf: nonCoreComponentNames) {
+                    return try process(provider, auxillaryPropertyMap: nonCoreComponentMap)
+                } else {
+                    return try process(provider, auxillaryPropertyMap: pluginExtensionMap)
+                }
+            } catch DependencyProviderContentError.propertyNotFound(let info) {
+                var message = "Could not find a provider for \(info.name): \(info.type) which was required by \(info.dependency)."
+                if let possibleMatchComponent = info.possibleMatchComponent {
+                    message += " Found possible matches \(info.possibleNames) at \(possibleMatchComponent)."
+                }
+                fatalError(message)
+            } catch {
+                fatalError("Unhandled error while processing dependency provider content: \(error)")
+            }
+        }
+    }
+
+    // MARK: - Private
+
+    struct AuxillaryProperties {
+        let sourceName: String
+        let properties: [Property]
+    }
+
+    private let providers: [DependencyProvider]
+    private let nonCoreComponentMap: [String: AuxillaryProperties]
+    private let pluginExtensionMap: [String: AuxillaryProperties]
+    private let nonCoreComponentNames: Set<String>
+
```

Comment:
It might be useful to add a comment saying the `auxillaryPropertyMap` key is the name of the pluginized component.

Reply:
done